update.py

```diff
@@ -1,58 +1,102 @@
 import os
 from os.path import dirname, join
 import sys
-from time import time
 import threading
 from traceback import print_exc
+import semantic_version
+from typing import Optional
+import tkinter as tk
```

Comment:
Importing tkinter *purely* to reference the type .Tk

Reply:
In this case a forward reference would be better I think (Simply because tk is a large import that we do nothing else with in this file). wrap the tkinter import in `if False` or `if typing.TYPE_CHECKING` and where you use it for types wrap them in strings.